packages/kbn-es/src/cluster.ts

```diff
@@ -389,7 +389,9 @@ export class Cluster {
     this.process = execa(ES_BIN, args, {
       cwd: installPath,
       env: {
-        ...(installPath ? { ES_TMPDIR: path.resolve(installPath, 'ES_TMPDIR') } : {}),
+        ...(installPath && process.env.FTR_DISABLE_ES_TMPDIR !== 'true'
+          ? { ES_TMPDIR: path.resolve(installPath, 'ES_TMPDIR') }
+          : {}),
```

Comment:
_question_: In the current implementation we explicitly check if `FTR_DISABLE_ES_TMPDIR` env is set in both `cluster.ts` and `install_archive.ts`.

What if we extend `InstallArchiveOptions` with new property ( e.g. `disableEsTempDir` and return together with `installPath`? Later we pass it together with `installPath` to `cluster.exec`. 

This way the code will be easier to debug and we can control the starting point where it is applied. Thoughts?

Reply:
Thanks, added in 405f845b2cfd3b902ada4c2cc9de1cc4ac36d88f. 

I didn't add to the CLI options because it seems like it is only needed in the FTR for now. If there is a need we can do another PR to extend it, and technically this setup has a bug which doesn't allow using the ES_TMPDIR env variable if someone wanted to override this behavior.

Reply:
Agree, no need to extend cli today. Thanks for adding comments and code change LGTM!
 
Integration tests are failing, sorry for extra work. Having a test to check `disableEsTempDir` is actually set to `true` might also be useful.

Reply:
@dmlemeshko I pushed 59a2eb1c18b37ad6e865ea87651533b56acac4e1. I tried to make a better test for the new code path, but since we need to mock `installArchive`, its not going to be hit.